hatch/src/cli/commands/mod.rs

```diff
@@ -48,3 +44,16 @@ pub trait Command<'command> {
     }
   }
 }
+
+fn parse_deps_from_cli<'func>(args: &ArgMatches<'func>) -> Vec<(String, String)> {
+  let mut parsed_deps = Vec::new();
+  if let Some(values) = args.values_of(INCLUDE) {
+    let mut vals = values.map(String::from).collect::<Vec<String>>().into_iter();
+    while vals.len() != 0 {
+      let mut url = vals.next();
+      let mut name = vals.next();
+      parsed_deps.push((url.unwrap(), name.unwrap()));
+    }
+  }
```

Comment:
how can we avoid the mutable `parsed_deps` and avoid the while loop? Can this be made more functional?

Reply:
Its only mutable in the scope of the function

Reply:
Yeah but it still feels like C/C++ pattern. I'm fine with it, but I would prefer to see a single line that produces the immutable collection using map/transform methods.

Reply:
Its worth revisiting later, theres a chance it will change before the completion of #45 but im fairly satisfied with it for now